### PR TITLE
Add TypeScript types

### PR DESCRIPTION
**Pull request by madhavarshney**
Added TypeScript types, ported from `ansi-colors`.

### CODE REVIEW

File: types/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+export const underline: KleurOrFunction;
+export const inverse: KleurOrFunction;
+export const hidden: KleurOrFunction;
+export const strikethrough: KleurOrFunction;
+
+/**
+ * Enable ANSI Colors.
+ * Note: You cannot write to ES6 exports and therefore to `kleur.enabled`
+ *       in TypeScript with `import * as kleur from 'kleur';`.
+ *       Workaround is to use `import kleur = require('kleur');`.
```
Comment (madhavarshney) [inline types/index.d.ts]: Do you want to add a function to set the `enabled` status as a work around?

**Comment by madhavarshney** [comment]
I removed the tests, please take a look.

File: package.json [hunk 2]
```diff
@@ -4,6 +4,8 @@
   "repository": "lukeed/kleur",
   "description": "The fastest Node.js library for formatting terminal text with ANSI colors~!",
   "license": "MIT",
+  "main": "./index.js",
+  "types": "./index.d.ts",
```
Comment (lukeed) [inline package.json]: Let's rename the file `kleur.d.ts`. 

Then this line should be `"types": "kleur.d.ts"`.

Lastly, have to add the file to the `"files"` list, or it won't be included in npm tarball.

```json
"files": [
  "*.js",
  "*.d.ts"
],
"types": "kleur.d.ts"
```

**Comment by lukeed** [comment]
Thank you~! Not a TS user myself, but someone will eventually need these lol

**Comment by madhavarshney** [comment]
I needed them, that's why I added the types 😄.

**Comment by AaronConlon** [comment]
Thanks.I needed them too.

**Pull request merged**